src/group_theory/subgroup.lean

```diff
@@ -1808,3 +1808,47 @@ set.subset.antisymm
 end subgroup
 
 end pointwise
+
+/-! Actions by subgroups -/
+section actions
+
+namespace subgroup
+
+variables {α β : Type*}
+
+/-- The action by a subgroup is the action by the underlying group. -/
+@[to_additive /-"The additive action by an add_subgroup is the action by the underlying
+add_group. "-/]
+instance [mul_action G α] (S : subgroup G) : mul_action S α :=
+mul_action.comp_hom _ S.subtype
+
+@[to_additive]
+lemma smul_def [mul_action G α] {S : subgroup G} (g : S) (m : α) : g • m = (g : G) • m := rfl
+
+@[to_additive]
+instance smul_comm_class_left
+  [mul_action G β] [has_scalar α β] [smul_comm_class G α β] (S : submonoid G) :
```

Comment:
```suggestion
  [mul_action G β] [has_scalar α β] [smul_comm_class G α β] (S : subgroup G) :
```

I'm guessing this is what was meant? Although I'm not sure why it needs to be defined separately for subgroups and submonoids.

Reply:
and idem down the file

Reply:
Shoot, that was careless, thanks. It has to be done for `subgroup` and `submonoid` because to lean those are completely different types. I suppose I could implement these as `S.to_submonoid.smul_comm_class_left` etc but it's not really any shorter.

Reply:
ahh, just played with structures on a test file; I didn't get how sub-structures work, and thought they got coe'd automatically if needed